### ISSUE TITLE
Change name of object, Position in get_gtfsrealtime?

### ISSUE THREAD

**Issue by hansthompson** [SymbolixAU/gtfsway#1]
Hey, this looks like a great package that I would put some time into helping with however I can.  

One thing I noticed that you might want to consider but doesn't necessarily break anything is that Position is already a base function and you write it as an object within get_gtfsrealtime.  Maybe rename to VehiclePosition? 

edit: okay, nm.  VehiclePosition is taken.

**Comment by SymbolixAU** [comment]
Thanks @hansthompson , good spot.

As this package is very much at its inception stage, I'm open to suggestions on functionality, what it should do, the output format, etc. So feel free to jump in with other comments or functions you think would be useful.

**Comment by hansthompson** [comment]
I'm not sure if you've seen this yet but I built a script that we run as a cron task in Anchorage, Alaska to create a real-time feed in R. 

https://github.com/codeforanchorage/api-realtime-bus/

It just does the basics though of presenting the trip update info.  

As for applications for this package, I think it should do some data analysis of real-time feeds.  

First order of business though is breaking the protobuf message into a list object.  I've had some trouble in the hourish I put in to break up the Aussie example you provided but I think it is the same problem across the board.  **edit: I can see you made a commit on this recently.** 

We'll have to dig a little deeper into the RProtobuf package to pull fields from messages.  The JSS article for the package has been the best documentation I've found. 

https://www.jstatsoft.org/article/view/v071i02

and of course the realtime.proto file is a useful resource for finding field names and data types. 

https://developers.google.com/transit/gtfs-realtime/gtfs-realtime-proto.

Do you have some thoughts on getting the message file into a list?  I was able to do it a while ago when I made the feed for the small example protobuf files so I'll start looking there to recreate what I did.

**Comment by SymbolixAU** [comment]
In my latest commit I've included a couple of functions that return lists of `data.tables` of trip updates. Is this sort of what you had in mind?

```
devtools::install_github("SymbolixAU/gtfsway")
library(gtfsway)
url <- "https://gtfsrt.api.translink.com.au/Feed/SEQ"

response <- httr::GET(url)

FeedMessage <- gtfs_realtime(response)
lst <- gtfs_tripUpdates(FeedMessage)
```

**Comment by hansthompson** [comment]
I get this message when using gtfs_tripUpdates (also the same one I get when playing with it myself); 

> [libprotobuf ERROR google/protobuf/wire_format.cc:1091] String field 'start_time' contains invalid UTF-8 data when parsing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes. 

I'm using window 7 and running R from a usb so I'll take a look further when I get home.  This happens for the protobuf I produce as well. 

And I haven't used data.table yet.  I would use data.frames but it should be up to you and I'll try to adapt to data.table.

**Comment by hansthompson** [comment]
At home, on ubuntu 16.04 I was able to use the snippet you provided but only with the Anchorage protobuf. 

```

library(gtfsway)
url <- "http://gtfs-rt.codeforanchorage.org/people_mover.pb"
response <- httr::GET(url)
FeedMessage <- gtfs_realtime(response)
lst <- gtfs_tripUpdates(FeedMessage)

```

I'm thinking commenting out some of the feed types until we can incrementally test each one would be a good way to proceed.  What do you think?

**Comment by SymbolixAU** [comment]
Yeah that's what I've been doing locally. From what I can gather, all the other feeds are subsets of `FeedMessage`

**Comment by XavierPrudent** [comment]
Hello, 
Your package seems to be relevant and interesting, may you please however describe in more details what you attempt to achieve?
Regards,
Xavier

**Comment by SymbolixAU** [comment]
hi @XavierPrudent , the 'details' are still to be defined at the moment. But as a starting point the ideas is for it to read a GTFS real-time feed that's defined using Google's protobuf format, and parse the message into an R object.

**Comment by XavierPrudent** [comment]
Hi, thanks for your answer. My question on SO was : assuming I downloaded the GTFS-rt protobuf file vehiclePositions.pb and TripUpdates.pb, can your code read them straight ahead? 
Like:
library(gtfsrealtime)
 vehiclePosition <- read.gtfsrt("vehiclePositions.pb", some.options)
?
Regards,
Xavier

**Comment by SymbolixAU** [comment]
short answer: no, it can't. 

The [`RProtoBuf` package](https://cran.r-project.org/web/packages/RProtoBuf/index.html) is probably what you're looking for.

**Comment by SymbolixAU** [comment]
@XavierPrudent  Do you have a link to those protobuf files so I can take a look at integrating them into the package?

**Comment by XavierPrudent** [comment]
Hello, sure, please find two of these files enclosed,
regards,
Xavier

[gtfs-protob.zip](https://github.com/SymbolixAU/gtfsway/files/856884/gtfs-protob.zip)
